solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -403,14 +403,9 @@ public void process(ResponseBuilder rb) throws IOException {
 
     req.getContext().put(SolrIndexSearcher.STATS_SOURCE, statsCache.get(req));
 
-    QueryResult result = new QueryResult();
-
     cmd.setSegmentTerminateEarly(
         params.getBool(
             CommonParams.SEGMENT_TERMINATE_EARLY, CommonParams.SEGMENT_TERMINATE_EARLY_DEFAULT));
-    if (cmd.getSegmentTerminateEarly()) {
-      result.setSegmentTerminatedEarly(Boolean.FALSE);
-    }
```

Comment:
observation: deferring this bit also helps with the `// not supported, silently ignore any segmentTerminateEarly flag` bit at lines 420/421 415/416 below i.e. if the results have `Boolean.FALSE` in future then it would not be despite the silent ignoring.

Reply:
Do you mean for grouping, which apparently doesn't support early termination?

Reply:
Yes. Currently with grouping the response would (via the presence of 'false') suggest that early termination was used but did not terminate early. With this change the 'segment terminated early' flag will be not present in the response, reflecting that early termination was not used (despite the user having asked for it).